### PR TITLE
Register Grid hooks for 1.7.8.0

### PR DESCRIPTION
**Pull request by matks**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 1.7.8.x
| Description?      | Register Grid hooks for 1.7.8.0
| Type?             | improvement
| Category?         | CO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | ~
| How to test?      | ~
| Possible impacts? | ~

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/24542)

### CODE REVIEW

File: install-dev/upgrade/sql/1.7.8.0.sql [hunk 1]
```diff
@@ -21,12 +21,51 @@ INSERT IGNORE INTO `PREFIX_hook` (`id_hook`, `name`, `title`, `description`, `po
   (NULL, 'actionFrontControllerSetVariables', 'Add variables in JavaScript object and Smarty templates', 'Add variables to javascript object that is available in Front Office. These are also available in smarty templates in modules.your_module_name.', '1'),
   (NULL, 'displayAdminGridTableBefore', 'Display before Grid table', 'This hook adds new blocks before Grid component table.', '1'),
   (NULL, 'displayAdminGridTableAfter', 'Display after Grid table', 'This hook adds new blocks after Grid component table.', '1'),
-  (NULL, 'displayAdminOrderCreateExtraButtons', 'Add buttons on the create order page dropdown', 'Add buttons on the create order page dropdown', '1')
```
Comment (matks) [inline install-dev/upgrade/sql/1.7.8.0.sql:24]: If I am not foolish these lines were missing the last comma 😮  so it was supposed to fail?

Comment (kpodemski) [reply install-dev/upgrade/sql/1.7.8.0.sql:24]: @matks https://github.com/PrestaShop/PrestaShop/pull/24242/files#diff-2ab29e497243cd89a7b49b94a35a33d9ca879e40ed9a3cf48853535ec304c423 😅

Comment (matks) [reply install-dev/upgrade/sql/1.7.8.0.sql:24]: Nice catch, looks like @atomiix was ahead of me one more time 😄 

We have a failing test from autoupgrade which indicates the issue => great! 💪 
https://github.com/PrestaShop/autoupgrade/actions/runs/855151733

But why only the 1.6 -> 1.7 runs have failed 🤔  . Such an obvious SQL issue should break everything?

**Review by Progi1984 — DISMISSED** [review]

File: install-dev/upgrade/sql/1.7.8.0.sql (hunk 1 above)
Comment (atomiix) [reply install-dev/upgrade/sql/1.7.8.0.sql:24]: @matks the failing doesn't come from the missing commas but I fixed them during the process of fixing the other issues

Comment (matks) [reply install-dev/upgrade/sql/1.7.8.0.sql:24]: @atomiix Then I'm scared 😨  if our autoupgrade CLI tests cannot detect a failing SQL statement

**Review by PierreRambaud — DISMISSED** [review]

**Comment by SD1982** [comment]
Hi @matks 
Can you please provide us at least a list of all the hook we have to check / test ?

**Comment by PierreRambaud** [comment]
imho, it doesn't require QA :thinking:

**Comment by matks** [comment]
> Hi @matks
> Can you please provide us at least a list of all the hook we have to check / test ?

There is 45 of them and I have 2 other PRs that will follow up with same quantity 😅  are you sure?

**Comment by Robin-Fischer-PS** [comment]
As seen with @PierreRambaud , it does not need manual QA.

Thanks @matks :)

File: install-dev/upgrade/sql/1.7.8.0.sql [hunk 2]
```diff
@@ -21,12 +21,52 @@ INSERT IGNORE INTO `PREFIX_hook` (`id_hook`, `name`, `title`, `description`, `po
...
-  (NULL, 'actionBeforeCreateProductFormHandler', 'Modify product identifiable object data before creating it', 'This hook allows to modify product identifiable object form data before it was created', '1')
-  (NULL, 'actionAfterCreateProductFormHandler', 'Modify product identifiable object data after creating it', 'This hook allows to modify product identifiable object form data after it was created', '1')
+  (NULL, 'displayAdminOrderCreateExtraButtons', 'Add buttons on the create order page dropdown', 'Add buttons on the create order page dropdown', '1'),
+  (NULL, 'actionProductFormBuilderModifier', 'Modify product identifiable object form', 'This hook allows to modify product identifiable object form content by modifying form builder data or FormBuilder itself', '1'),
+  (NULL, 'actionBeforeUpdateProductFormHandler', 'Modify product identifiable object data before updating it', 'This hook allows to modify product identifiable object form data before it was updated', '1'),
+  (NULL, 'actionAfterUpdateProductFormHandler', 'Modify product identifiable object data after updating it', 'This hook allows to modify product identifiable object form data before it was updated', '1'),
```
Comment (PululuK) [inline install-dev/upgrade/sql/1.7.8.0.sql]:
```suggestion
  (NULL, 'actionAfterUpdateProductFormHandler', 'Modify product identifiable object data after updating it', 'This hook allows to modify product identifiable object form data after it was updated', '1'),
```

Comment (PululuK) [reply install-dev/upgrade/sql/1.7.8.0.sql]: See https://github.com/PrestaShop/PrestaShop/pull/24847

**Comment by Progi1984** [comment]
@matks Need rebase

File: install-dev/upgrade/sql/1.7.8.0.sql (hunk 2 above)
Comment (matks) [reply install-dev/upgrade/sql/1.7.8.0.sql]: Indeed

**Comment by matks** [comment]
Work continued and finished in PR https://github.com/PrestaShop/PrestaShop/pull/24885

**Pull request closed without merging**